http/src/main/java/com/hazelcast/jet/contrib/http/HttpListenerBuilder.java

```diff
@@ -0,0 +1,192 @@
+        this.sslContextFn = Objects.requireNonNull(sslContextFn);
+        return this;
+    }
+
+    /**
+     * Set the function which converts the received payload to the pipeline
+     * item.
+     * <p>
+     * For example to convert the payload to a {@link String}:
+     * <pre>{@code
+     * builder.mapToItemFn(data -> new String(data))
+     * }</pre>
+     * <p>
+     * By default payload is converted to String. If this parameter is set
+     * {@link #type(Class)} is ignored.
+     *
+     * @param mapToItemFn the function which converts the received payload to
+     *                    pipeline item.
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public <T_NEW> HttpListenerBuilder<T_NEW> mapToItemFn(@Nonnull FunctionEx<byte[], T_NEW> mapToItemFn) {
+        HttpListenerBuilder<T_NEW> newThis = (HttpListenerBuilder<T_NEW>) this;
+        newThis.mapToItemFn = Objects.requireNonNull(mapToItemFn);
+        return newThis;
+    }
+
+    /**
+     * Set the type of the object which the source will map the JSON formatted
+     * payload. The payload is converted to {@link String} first and then
+     * mapped to the given type.
+     * <p>
+     * For example to convert the payload to a person object:
+     * <pre>{@code
+     * builder.type(Person.class)
+     * }</pre>
+     * <p>
+     * Default value is {@code null}. If {@link #mapToItemFn(FunctionEx)} is
+     * set this parameter is ignored.
+     *
+     * @param type the type of the object which the source will map the JSON
+     *             formatted payload.
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public <T_NEW> HttpListenerBuilder<T_NEW> type(@Nonnull Class<T_NEW> type) {
+        HttpListenerBuilder<T_NEW> newThis = (HttpListenerBuilder<T_NEW>) this;
+        newThis.type = Objects.requireNonNull(type);
+        return newThis;
+    }
+
+    /**
+     * Build a HTTP(s) Listener {@link StreamSource} with the supplied
+     * parameters.
+     */
+    @Nonnull
+    public StreamSource<T> build() {
+        int localPort = port;
+        FunctionEx<byte[], T> mapFn = mapFn();
+        SupplierEx<SSLContext> sslFn = sslContextFn;
+
+        return SourceBuilder.stream(name(),
+                ctx -> new HttpListenerSourceContext<>(ctx, localPort, sslFn, mapFn))
+                .<T>fillBufferFn(HttpListenerSourceContext::fillBuffer)
+                .destroyFn(HttpListenerSourceContext::close)
+                .distributed(1)
+                .build();
+    }
+
+    private String name() {
+        if (sslContextFn == null) {
+            return "http-listener[" + port + "]";
+        }
+        return "https-listener[" + port + "]";
+    }
+
+    @SuppressWarnings("unchecked")
+    private FunctionEx<byte[], T> mapFn() {
+        if (mapToItemFn == null) {
+            if (type != null) {
+                Class<T> theType = type;
+                return data -> JsonUtil.beanFrom(new String(data), theType);
```

Comment:
What's the purpose of `theType` here? `type` can be used directly. Maybe leftover from before this was part of the `build()` method to avoid serialization of the builder instance?

Reply:
yes, leftover

Reply:
since we use it in a lambda it captures the whole class, by creating a local copy we detach the variable from the class.